internal/exec/exec.go

```diff
@@ -303,6 +303,9 @@ func (r *request) addError(err *errors.QueryError) {
 func (r *request) handlePanic() {
 	if err := recover(); err != nil {
 		execErr := errors.Errorf("graphql: panic occured: %v", err)
+		if err, ok := err.(error); ok {
+			execErr.ResolverError = err
```

Comment:
If you want access to the originating error to extract information from it, then the error is not unexpected and thus it does not qualify for a panic, right? ;-)